publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/service/EventTemplateHandler.java

```diff
@@ -44,19 +48,20 @@ public class EventTemplateHandler {
 
     private static final String REGEXP_END_DIGITS = "\\[\\d+\\]$";
 
-    private final Configuration configuration = Configuration.builder()
-            .jsonProvider(new JacksonJsonNodeJsonProvider())
-            .mappingProvider(new JacksonMappingProvider())
-            .build();
+    private ObjectMapper mapper = JsonMapper.builder()
+        .disable(JsonNodeFeature.READ_NULL_PROPERTIES)
+        .build()
+            .setSerializationInclusion(JsonInclude.Include.NON_EMPTY);
+    private Configuration configuration = Configuration.builder()
+        .jsonProvider(new JacksonJsonNodeJsonProvider(mapper))
+        .mappingProvider(new JacksonMappingProvider(mapper))
+        .build();
 
     // eventTemplateParser
+    @JsonInclude(JsonInclude.Include.NON_NULL)
     public JsonNode eventTemplateParser(String jsonData , String eventName){
         JsonNode updatedJson = null;
-        JsonFactory factory = new JsonFactory();
-        ObjectMapper mapper = new ObjectMapper(factory);
```

Comment:
Why to refactor its same as old code

Reply:
what about NON_EMPTY ?

Reply:
> Why to refactor its same as old code

mapper is instance variable now, used to create/initialize configuration variable, used in several different methods. That's why it changed from local var to instance var.

Reply:
> what about NON_EMPTY ?

see line 54

Reply:
Without this change tests fail with errors like this:

```
[ERROR]   EventTemplateHandlerTest.testEventParserWithEventEiffelTestSuiteFinishedEvent:234->testParser:369 eventParams.links[target=e269b37d-17a1-4a10-aafb-c108735ee48c]
Unexpected: domainId
 ; eventParams.links[target=e269b37d-17a1-4a10-aafb-c108735ee48d]
Unexpected: domainId

[ERROR] Tests run: 50, Failures: 28, Errors: 0, Skipped: 0
```